src/Core/SymbolUtility.cs

```diff
@@ -695,4 +709,52 @@ public static bool IsAwaitable(ITypeSymbol typeSymbol, bool shouldCheckWindowsRu
 
         return typeSymbol as INamedTypeSymbol;
     }
+
+    /// <summary>
+    /// Determines if the type is a task type - that is, <see cref="Task"/>, <see cref="Task{TResult}"/>, or a <see href="https://github.com/dotnet/roslyn/blob/main/docs/features/task-types.md">task-like type</see>.<br/>
+    /// Only task types (and <see langword="void"/>) can be the return type of a method marked with the <see langword="async"/> keyword.
+    /// </summary>
+    /// <remarks>
+    /// For more information, see the <see href="https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#15151-general">C# language specification</see>.
+    /// </remarks>
+    /// <returns>A <see cref="bool"/> indicating whether the type is <see cref="Task"/>, <see cref="Task{TResult}"/>, or a <see href="https://github.com/dotnet/roslyn/blob/main/docs/features/task-types.md">task-like type</see>.</returns>
+    public static bool IsTaskLikeType(this ITypeSymbol typeSymbol)
```

Comment:
This method is used only once and that's why I would just inline it into `IsAwaitableTaskType` + if it's inlined we know that condition

```cs
if (definition.HasMetadataName(in MetadataNames.System_Threading_Tasks_Task)
    || definition.HasMetadataName(in MetadataNames.System_Threading_Tasks_Task_T))
{
    return true;
}
```

is redundant (it will never succeed).